src/helper/todoist.ts

```diff
@@ -30,11 +30,18 @@ export const pullTask = async () => {
   const { preferredDateFormat } = await logseq.App.getUserConfigs()
   const { todoist } = settings
   let tasks: Task[] = []
+
   if (todoist?.sync === 1 && todoist.project) {
-    tasks = await instance.getTasks({ projectId: todoist.project })
+    tasks = await instance.getTasks({ projectId: todoist.project });
+  } else if (todoist?.sync === 2 && todoist.filter) {
+    if (!todoist.filter) {
+      console.error("Missing filter, cannot sync using the filter sync option");
```

Comment:
Perhaps we can make this notification visible to users as well.
`logseq.UI.showMsg('Missing filter, cannot sync using the filter sync option', 'error')`

Reply:
I like it. I have made the change and for consistency applied it to the project sync as well.